core/legacy/config/config.js

```diff
@@ -128,29 +128,6 @@ function assertValidGatherer(gathererInstance, gathererName) {
   }
 }
 
-
-/**
- * Validate the LH.Flags
- * @param {LH.Flags} flags
- */
-function assertValidFlags(flags) {
-  // COMPAT: compatibility layer for devtools as it uses the old way and we need tests to pass
-  // TODO(paulirish): remove this from LH once emulation refactor has rolled into DevTools
-  // @ts-expect-error Deprecated flag
-  if (flags.channel === 'devtools' && flags.internalDisableDeviceScreenEmulation) {
-    // @ts-expect-error Deprecated flag
-    flags.formFactor = flags.emulatedFormFactor;
-    // @ts-expect-error Deprecated flag
-    flags.emulatedFormFactor = flags.internalDisableDeviceScreenEmulation = undefined;
-  }
-
-
-  // @ts-expect-error Checking for removed flags
-  if (flags.emulatedFormFactor || flags.internalDisableDeviceScreenEmulation) {
-    throw new Error('Invalid emulation flag. Emulation configuration changed in LH 7.0. See https://github.com/GoogleChrome/lighthouse/blob/main/docs/emulation.md');
```

Comment:
@paulirish should we keep?

Reply:
FWIW there isn't an equivalent check in FR